### PR TITLE
Sv 889 handle reapplies for withdrawn standard admin svc

### PR DESCRIPTION
**Pull request by anilkdhiman6**

### CODE REVIEW

**Review by CodeWolf86 — CHANGES_REQUESTED** [review]
There are no unit tests added for this against the ApplicationController. It's one of the key controllers in the admin service so I'd like to see some against the new code.

File: src/SFA.DAS.AdminService.Web/ViewModels/Apply/Applications/SequenceViewModel.cs [hunk 1]
```diff
@@ -41,6 +43,40 @@ public class SequenceViewModel : BackViewModel
...
+                //Version applications
+                if (application.StandardApplicationType == StandardApplicationTypes.Version)
+                {
+                    foreach (var withdrawal in previousWithdrawals)
+                    {
+                        foreach (var ver in withdrawal.ApplyData.Apply.Versions)
+                        {
+                            if (withdrawal.ApplyData.Apply.Versions.Contains(ver))
+                                return withdrawal;
+                        }
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.AdminService.Web/ViewModels/Apply/Applications/SequenceViewModel.cs:64]: Yes - this is a typo - should be:

```
     if (application.ApplyData.Apply.Versions.Contains(ver))
           return withdrawal;
```

However have refactored to improve readability.

File: src/SFA.DAS.AdminService.Web/ViewModels/Apply/Applications/SequenceViewModel.cs [hunk 2]
```diff
@@ -41,6 +43,40 @@ public class SequenceViewModel : BackViewModel
...
+                    }
+                }
+                else //Standard applications
+                {
+                    foreach (var withdrawal in previousWithdrawals)
+                    {
+                        if (withdrawal.ApplicationType == StandardApplicationTypes.StandardWithdrawalApp)
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.AdminService.Web/ViewModels/Apply/Applications/SequenceViewModel.cs:71]: Previous Withdrawals are returned ordered by **CreatedAt** from the SQL Query.

Have refactored this method

File: src/SFA.DAS.AdminService.Web/Controllers/Apply/ApplicationController.cs [hunk 3]
```diff
@@ -450,36 +452,47 @@ public async Task<IActionResult> Return(Guid applicationId, int sequenceNo, stri
             }
 
             var warningMessages = new List<string>();
-            if (sequenceNo == ApplyConst.STANDARD_SEQUENCE_NO && returnType == ReturnTypes.Approve)
+            var withdrawal = await _applyApiClient.GetWithdrawnApplications(application.OrganisationId, application.StandardCode);
```
[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.AdminService.Web/Controllers/Apply/ApplicationController.cs:455]: Can we catch up regarding the logic for this please  @CodeWolf86

[the remark this answers is not in the document]
Comment (anilkdhiman6) [reply src/SFA.DAS.AdminService.Web/Controllers/Apply/ApplicationController.cs]: Refactored - altered as discussed

**Review by CodeWolf86 — APPROVED** [review]

**Review by CodeWolf86 — APPROVED** [review]

**Pull request merged**